### PR TITLE
Add database view for UpdatesScreen

### PR DESCRIPTION
**Pull request by ghostbear**
This PR also adds selection interactions and navigation to manga detail page when clicking on the cover

https://user-images.githubusercontent.com/6576096/128592454-e029f749-750e-4a7d-a35f-a0edc6f2a551.mp4

### CODE REVIEW

File: presentation/src/main/kotlin/updates/UpdatesScreen.kt [hunk 1]
```diff
@@ -96,3 +135,11 @@ fun UpdatesItem(
     }
   }
 }
+
+private fun Modifier.selectionBackground(isSelected: Boolean): Modifier = composed {
```
Comment (arkon) [inline presentation/src/main/kotlin/updates/UpdatesScreen.kt:139]: We'll probably end up needing this elsewhere.

File: data/src/main/kotlin/updates/model/UpdatesManga.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        manga.id, manga.sourceId, manga.`key`, manga.title, manga.cover, manga.favorite, 
+        chapter.dateUpload, chapter.id chapterId, chapter.name, chapter.read, chapter.number, 
+        date(ROUND(chapter.dateUpload / 1000), 'unixepoch', 'localtime') date
+    FROM manga 
+    INNER JOIN chapter ON manga.id = chapter.mangaId
+    WHERE manga.favorite = 1
+    """,
+  viewName = "updates"
+)
+data class UpdatesManga(
```
[the remark this answers is not in the document]
Comment (ghostbear) [reply data/src/main/kotlin/updates/model/UpdatesManga.kt:28]: Think all models current are duplicated like this. I like the idea because you have one clean data class without annotations which kinda makes it easier to read, imo. Sure, this isn't a data class with a lot of annotation so it is not hard to follow what it does.

@inorichi is there an actual reason why they are split like this?

File: presentation/src/main/kotlin/updates/UpdatesScreen.kt [hunk 3]
```diff
@@ -51,30 +62,58 @@ fun UpdatesScreen(navController: NavController) {
       )
     }
   ) {
+    LazyColumn(
+      contentPadding = PaddingValues(top = 8.dp, bottom = 72.dp)
```
[the remark this answers is not in the document]
Comment (ghostbear) [reply presentation/src/main/kotlin/updates/UpdatesScreen.kt:66]: Discovered that the accompanist library which contains insetter can do this for us

File: presentation/src/main/kotlin/core/util/DateExtensions.kt [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package tachiyomi.ui.core.util
+
+import android.text.format.DateUtils
+import java.util.Date
```
[the remark this answers is not in the document]
Comment (arkon) [reply presentation/src/main/kotlin/core/util/DateExtensions.kt:4]: #23

File: data/src/main/kotlin/updates/model/UpdatesManga.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (inorichi) [reply data/src/main/kotlin/updates/model/UpdatesManga.kt:28]: Because we can't use Room annotations on the domain module because it's a Java module. Even if it were an Android module, we'd tie a database implementation to the domain module which should be platform-agnostic.

The clean architecture pattern usually declares a domain class, a data class and a mapper to convert between them. However, I wanted to avoid this small overhead by using the same class instance, so Room already creates domain classes without any mapper required, but to avoid package collisions, I added a gradle task in the data module to remove those duplicated classes after Room has generated their mappers.

I also considered making the domain classes interfaces, but then we'd lose some of the handy methods of `data class`es like `copy`, although it probably wouldn't be a big loss.

[the remark this answers is not in the document]
Comment (ghostbear) [reply data/src/main/kotlin/updates/model/UpdatesManga.kt:28]: Okay, I see, thanks for the clarification.

> I also considered making the domain classes interfaces, but then we'd lose some of the handy methods of data classes like copy, although it probably wouldn't be a big loss.

This would also help with better error messages when forgetting to match the models when one changes a variable type or adding/removing a variable

[the remark this answers is not in the document]
Comment (inorichi) [reply data/src/main/kotlin/updates/model/UpdatesManga.kt:28]: I can try to explore turning the domain classes into interfaces. I wouldn't need the gradle task hack then. But I don't know if I'm missing something and I'm going to hit a wall I'm not expecting :sweat_smile:

**Comment by inorichi** [comment]
LGTM. I'd create an `UpdatesState` like I'm doing for the catalogs to separate state from the view model, but it can be done later.
